apps/dashboard/app/models/project.rb

```diff
@@ -23,6 +23,34 @@ def lookup_table
       {}
     end
 
+    def from_directory(dir)
```

Comment:
Can you move this to a class method instead of an instance method?

Reply:
I am confused; it is a class method right now, as it is written under `class << self`.

Reply:
> I am confused; it is a class method right now, as it is written under class << self.

It's an instance method right now. A class method is a 'static function' in other languages.